operator/internal/lifecycle/testdata/cases.pools.golden.txtar

```diff
@@ -358,6 +358,831 @@
+          startupProbe:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - |
+                set -e
+                RESULT=$(curl --silent --fail -k -m 5 --cacert /etc/tls/certs/default/ca.crt "https://${SERVICE_NAME}.compat-test.compat-test.svc.cluster.local.:9644/v1/status/ready")
+                echo $RESULT
+                echo $RESULT | grep ready
+            failureThreshold: 120
+            initialDelaySeconds: 1
+            periodSeconds: 10
+          volumeMounts:
+          - mountPath: /etc/tls/certs/default
+            name: redpanda-default-cert
+          - mountPath: /etc/tls/certs/external
+            name: redpanda-external-cert
+          - mountPath: /etc/redpanda
+            name: config
+          - mountPath: /tmp/base-config
+            name: base-config
+          - mountPath: /var/lifecycle
+            name: lifecycle-scripts
+          - mountPath: /var/lib/redpanda/data
+            name: datadir
+          - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+            name: kube-api-access
+            readOnly: true
+        - args:
+          - supervisor
+          - --
+          - /redpanda-operator
+          - sidecar
+          - --redpanda-yaml
+          - /etc/redpanda/redpanda.yaml
+          - --redpanda-cluster-namespace
+          - compat-test
+          - --redpanda-cluster-name
+          - compat-test
+          - --run-broker-probe
+          - --broker-probe-broker-url
+          - $(SERVICE_NAME).compat-test.compat-test.svc.cluster.local.:9644
+          command:
+          - /redpanda-operator
+          env:
+          - name: SERVICE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: POD_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          - name: HOST_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.hostIP
+          image: localhost/redpanda-operator:dev
+          name: sidecar
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /healthz
+              port: 8093
+            initialDelaySeconds: 1
+            periodSeconds: 10
+            successThreshold: 1
+          resources:
+            requests:
+              cpu: 16m
+              memory: 64Mi
+          volumeMounts:
+          - mountPath: /etc/tls/certs/default
+            name: redpanda-default-cert
+          - mountPath: /etc/tls/certs/external
+            name: redpanda-external-cert
+          - mountPath: /etc/redpanda
+            name: config
+          - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+            name: kube-api-access
+            readOnly: true
+        initContainers:
+        - command:
+          - /bin/bash
+          - -c
+          - rpk redpanda tune all
+          image: foo:bar
```

Comment:
I don't understand why this `foo:bar` is rendered if you changed initContainer to `public.ecr.aws/docker/library/busybox:1.36-glibc@sha256:e046063223f7eaafbfbc026aa3954d9a31b9f1053ba5db04a4f1fdc97abd8963`. Only `set-datadir-ownership` is set with that, why?

Reply:
because pretty much all of the rendering code in our chart leverage `Values.Image` (the Redpanda image) for anything that shells out to rpk in an init container: https://github.com/redpanda-data/redpanda-operator/blob/2c26dd5913b8851e9d3c1762f75c2f981288f833/charts/redpanda/statefulset.go#L364

whereas the `set-datadir-ownership` container comes from `Statefulset.InitContainerImage`: https://github.com/redpanda-data/redpanda-operator/blob/2c26dd5913b8851e9d3c1762f75c2f981288f833/charts/redpanda/statefulset.go#L397

The image overrides on nodepools basically correspond to:

- `NodePool.Image` --> override of the global `Values.Image`
- `NodePool.InitContainerImage` --> equivalent `Statefulset.InitContainerImage`
- `NodePool.SidecarImage` --> equivalent `Statefulset.SideCars.Image`